src/features/authentication/login/login.tsx

```diff
@@ -18,138 +17,240 @@ import "./login.scss";
 import { observer } from "mobx-react-lite";
-
 interface LoginFormProps {
   loginStore: LoginStoreImplementation;
 }
 
+interface UserData {}
+
 export const LoginForm: React.FC<LoginFormProps> = observer(
   ({ loginStore }) => {
+    const navigate = useNavigate();
+
     const [checked, setChecked] = useState(false);
+    const [user, setUser] = useState<UserData | null>(null);
 
-    const [email, setEmail] = useState("");
-    const [password, setPassword] = useState("");
-    const loginToApp = (event: any) => {
-      event.preventDefault();
-
-      signInWithEmailAndPassword(auth, email, password)
-        .then(() => {
-          loginStore.loginSucces(onAuthStateChanged);
-        })
-        .catch((err) => {
-          console.log(email, password);
-        });
+    const initialValues = {
+      email: "",
+      password: "",
+    };
+    const [formValues, setFormValues] = useState(initialValues);
+    const [formErrors, setFormErrors] = useState(initialValues);
+    const [isSubmit, setIsSubmit] = useState(false);
+
+    const loginToApp = async () => {
+      const errors = initialValues;
+      try {
+        const user = await signInWithEmailAndPassword(
+          auth,
+          formValues.email,
+          formValues.password
+        );
+        localStorage.setItem("authenticated", JSON.stringify(true));
+        navigate("/homepage");
+      } catch (error: any) {
+        console.log(error.message);
+        if (error.message.includes("user-not-found")) {
+          errors.email = "Email is incorrect.";
+          setFormErrors(errors);
+        } else if (error.message.includes("password")) {
+          errors.password = "Password is incorrect.";
+          setFormErrors(errors);
+        }
+      }
     };
 
-    const register = () => {
-      createUserWithEmailAndPassword(auth, email, password)
-        .then((userAuth) => {
-          console.log("User created");
-        })
-        .catch((err) => {
-          alert(err);
-        });
+    const signInWithGoogle = async () => {
+      const provider = new GoogleAuthProvider();
+      try {
+        await signInWithPopup(auth, provider);
+        localStorage.setItem("authenticated", JSON.stringify(true));
+        navigate("/homepage");
+      } catch (error: any) {
+        console.log(error.message);
+      }
+    };
+
+    const signInWithFacebook = async () => {
+      const provider = new FacebookAuthProvider();
+      try {
+        await signInWithPopup(auth, provider);
+        localStorage.setItem("authenticated", JSON.stringify(true));
+        navigate("/homepage");
+      } catch (error: any) {
+        console.log(error.message);
+      }
+    };
+
+    const logout = async () => {
+      await signOut(auth);
+      localStorage.setItem("authenticated", JSON.stringify(false));
+    };
+
+    const handleChange = (e: any) => {
+      const { name, value } = e.target;
+      setFormValues({ ...formValues, [name]: value });
```

Comment:
In order to make sure that the state from the right context is picked up, this can be written as: `setFormValues((prevState) => ({ ...prevState, [name]: value }))`

Reply:
Oh, I didn't know about this prevState method, nice.